config/default-spinnaker-local.yml

```diff
@@ -110,9 +109,11 @@ providers:
     enabled: ${SPINNAKER_KUBERNETES_ENABLED:false}
 
     primaryCredentials:
-      name: my-docker-registry-account
-      address: https://index.docker.io/
-      repository: library/nginx
+      name: my-docker-registry
+      address: ${SPINNAKER_DOCKER_REGISTRY:https://index.docker.io/}
+      repository: ${SPINNAKER_DOCKER_REPOSITORY:library/nginx}
+      username: ${SPINNAKER_DOCKER_USERNAME}
+      passwordFile: ${SPINNAKER_DOCKER_PASSWORD_FILE}
```

Comment:
What is a passwordFile? Is this standard? Is it keyed or flat?

You might want to modify reconfigure_spinnaker to check permissions on it.